internal/utils/miscellaneous.go

```diff
@@ -81,3 +82,33 @@ func ComputeScheduleInterval(cronSchedule string) (time.Duration, error) {
 	nextNextScheduledTime := schedule.Next(nextScheduledTime)
 	return nextNextScheduledTime.Sub(nextScheduledTime), nil
 }
+
+func getValueOrError(data map[string]string, key string) (string, error) {
+	value, ok := data[key]
+	if !ok {
+		return "", fmt.Errorf("key %s not found in data", key)
+	}
+	return value, nil
+}
+
+func parseBoolOrError(value string) (bool, error) {
+	result, err := strconv.ParseBool(value)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse boolean value: %w", err)
+	}
+	return result, nil
+}
+
+// GetBoolValueOrDefault returns the boolean value for the given key from the data map,
+// and defaults to defaultValue if the key is not found or the value is not a valid boolean.
+func GetBoolValueOrDefault(data map[string]string, key string, defaultValue bool) bool {
```

Comment:
if `parseBoolOrError` returns an error then is it right to use the default value? You are effectively hiding a programming bug.

Reply:
We rely on `strconv.FormalBool()` and `strconv.ParseBool()` to convert between string and bool values, so I'm assuming that we should not be encountering any errors within `GetBoolValueOrDefault()` function. If you want, I can return the error and propagate it back up, which will result in a requeue of reconciliation. WDYT?

Reply:
yes eating away errors is not good and points to a programming error which should surface and be fixed.

Reply:
Addressed in https://github.com/gardener/etcd-druid/pull/996/commits/29e8a87f8edb87c35fe9144805e30b68d0cbda8e